azurerm/resource_arm_storage_data_lake_gen2_filesystem.go

```diff
@@ -0,0 +1,225 @@
+			return tf.ImportAsExistsError("azurerm_storage_data_lake_gen2_filesystem", id)
+		}
+	}
+
+	log.Printf("[INFO] Creating File System %q in Storage Account %q.", fileSystemName, storageID.Name)
+	input := filesystems.CreateInput{
+		Properties: properties,
+	}
+	if _, err := client.Create(ctx, storageID.Name, fileSystemName, input); err != nil {
+		return fmt.Errorf("Error creating File System %q in Storage Account %q: %s", fileSystemName, storageID.Name, err)
+	}
+
+	d.SetId(id)
+	return resourceArmStorageDataLakeGen2FileSystemRead(d, meta)
+}
+
+func resourceArmStorageDataLakeGen2FileSystemUpdate(d *schema.ResourceData, meta interface{}) error {
+	accountsClient := meta.(*ArmClient).Storage.AccountsClient
+	client := meta.(*ArmClient).Storage.FileSystemsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := filesystems.ParseResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	storageID, err := storage.ParseAccountID(d.Get("storage_account_id").(string))
+	if err != nil {
+		return err
+	}
+
+	// confirm the storage account exists, otherwise Data Plane API requests will fail
+	storageAccount, err := accountsClient.GetProperties(ctx, storageID.ResourceGroup, storageID.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(storageAccount.Response) {
+			return fmt.Errorf("Storage Account %q was not found in Resource Group %q!", storageID.Name, storageID.ResourceGroup)
+		}
+
+		return fmt.Errorf("Error checking for existence of Storage Account %q (Resource Group %q): %+v", storageID.Name, storageID.ResourceGroup, err)
+	}
+
+	propertiesRaw := d.Get("properties").(map[string]interface{})
+	properties := storage.ExpandMetaData(propertiesRaw)
+
+	log.Printf("[INFO] Updating Properties for File System %q in Storage Account %q.", id.DirectoryName, id.AccountName)
+	input := filesystems.SetPropertiesInput{
+		Properties: properties,
+	}
+	if _, err = client.SetProperties(ctx, id.AccountName, id.DirectoryName, input); err != nil {
+		return fmt.Errorf("Error updating Properties for File System %q in Storage Account %q: %s", id.DirectoryName, id.AccountName, err)
+	}
+
+	return resourceArmStorageDataLakeGen2FileSystemRead(d, meta)
+}
+
+func resourceArmStorageDataLakeGen2FileSystemRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Storage.FileSystemsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := filesystems.ParseResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// TODO: what about when this has been removed?
+	resp, err := client.GetProperties(ctx, id.AccountName, id.DirectoryName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] File System %q does not exist in Storage Account %q - removing from state...", id.DirectoryName, id.AccountName)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving File System %q in Storage Account %q: %+v", id.DirectoryName, id.AccountName, err)
+	}
+
+	d.Set("name", id.DirectoryName)
+
```

Comment:
Can we set `storage_account_id` here?

Reply:
nope - it's not returned (hence the import function)